cashctrl_ledger/ledger.py

```diff
@@ -47,11 +77,71 @@ def add_price():
         """
         raise NotImplementedError
 
-    def add_vat_code():
+    def add_vat_code(
+        self, code: str, rate: float, account: str, inclusive: bool = True,
+        text: str = ""
+    ):
         """
-        Not implemented yet
+        Adds a new VAT code to the CashCtrl account.
+
+        Parameters:
+            code (str): The VAT code to be added.
+            rate (float): The VAT rate, must be between 0 and 1.
+            account (str): The account identifier to which the VAT is applied.
+            inclusive (bool): Determines whether the VAT is calculated as 'NET' 
+                            (True, default) or 'GROSS' (False).
+            text (str): Additional text or description associated with the VAT code.
         """
-        raise NotImplementedError
+        accounts = self._client.list_accounts()
+        account_map = accounts.set_index('number')['id'].to_dict()
+        payload = {
+            "name": code,
+            "percentage": rate*100,
+            "accountId": account_map.get(account, None),
+            "calcType": "NET" if inclusive else "GROSS",
+            "documentName": text,
+        }
+
+        self._client.post("tax/create.json", data=payload)
+        
+    def update_vat_code(
+        self, code: str, rate: float, account: str,
+        inclusive: bool = True, text: str = ""
+    ):
+        """
+        Updates an existing VAT code in the CashCtrl account with new parameters. 
+
+        Parameters:
+            code (str): The VAT code to be updated.
+            rate (float): The VAT rate, must be between 0 and 1.
+            account (str): The account identifier to which the VAT is applied.
+            inclusive (bool): Determines whether the VAT is calculated as 'NET' 
+                            (True, default) or 'GROSS' (False).
+            text (str): Additional text or description associated with the VAT code.
+        """
+
+        accounts = self._client.list_accounts()
+        vat_account = accounts.loc[accounts['number'] == account]
+
+        if not vat_account.empty:
+            account = vat_account['id'].iloc[0]
+        else:
+            account = None
+
+        remote_vats = self._client.list_tax_rates()
+        remote_vat = remote_vats.loc[remote_vats['name'] == code]
+        remote_vat_id = remote_vat['id'].iloc[0] if not remote_vat.empty else None
```

Comment:
- Please ensure there is exactly one `remote_vat` entry before accessing item zero.
- Please drop the else part. The update method does not make any sense, if no remote_vat is found.